### ISSUE TITLE
`manual_expect` fixer returns not compilable code.

### ISSUE THREAD

**Issue by wawel37** [software-mansion/cairo-lint#365]
### Problem

https://github.com/software-mansion/cairo-lint/blob/58a224c3afb39a363b36ec176b65e4043af6a3cb/tests/manual/manual_expect.rs#L434. This test generates code that doesn't compile.

### Scarb Version

```text
2.11.4
```

**Comment by Arcticae** [comment]
From my research it seems that the following code (from the test), should not even be caught by the lint:
```cairo
fn main() {
    let res_val: Result<i32> = Result::Err('err');
    // This is just a variable.
    let _a = match res_val {
        Result::Ok(val) => val,
        Result::Err(err) => core::panic_with_felt252(err)
    };
}
```
It's more a case for "manual `unwrap_or_else`" lint, as it can be substituted with `unwrap_or_else`. There is no code with `expect` that would work in that case, thus it should not be detected with this lint. 

So therefore - the correct solution is to omit detecting this case, and detect it with another lint
